src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -150,6 +150,12 @@
 #  define USE_SOF     0
 #endif
 
+// Some STM32F3 devices allow to remap the USB interrupt vectors from
+// shared USB/CAN IRQs to separate CAN and USB IRQs
+#ifndef STM32F3_IRQ_REMAP
```

Comment:
This is interesting, I didn't know about this, could you tell me more about the specific MCU that support this, and which section in the reference manual for this. I would like to read this up a bit, to see if we could have a better systematic solution. I kind of not wanting to have custom macro for specific DCD if there is a better way.

Reply:
This only exists for the [STM32F302 and the larger STM32F303 devices](https://github.com/modm-io/modm-devices/search?q=USBWakeUp_RMP).

See [RM0365 Table 40, page 208, footnote 1](https://www.st.com/resource/en/reference_manual/dm00094349-stm32f302xbcde-and-stm32f302x68-advanced-armbased-32bit-mcus-stmicroelectronics.pdf#page=208) (STM32F302) and [RM0318 Table 82, page 288, footnote 1](https://www.st.com/resource/en/reference_manual/dm00043574-stm32f303xb-c-d-e-stm32f303x6-8-stm32f328x8-stm32f358xc-stm32f398xe-advanced-arm-based-mcus-stmicroelectronics.pdf#page=288) (STM32F303xb/c/d/e):

> 1. It is possible to remap the USB interrupts (USB_HP, USB_LP and USB_WKUP) on interrupt lines 74, 75 and 76 respectively by setting the USB_IT_RMP bit in the Section 11.1.1: SYSCFG configuration register 1 (SYSCFG_CFGR1) on page 172.

Reply:
I guess you could read the bit at runtime and select the right interrupts. Wouldn't be too much overhead.

Reply:
@salkinium sorry for late response, this is great idea to solve the shared CAN/USB. I like your last suggestion by checking the actual remap bit. The overhead is too small ( a couple of cycle for bit test) and this aren't call at any fast rate either. Would you mind changing it to dynamic check. Also since it only available to selected chips, it should be wrapped with `SYSCFG_CFGR1_USB_IT_RMP` e.g

```
#ifdef SYSCFG_CFGR1_USB_IT_RMP
if ( remapped )
{
 use RMP IRQ
}else
#endif
{
  use normal IRQ 
}
```

Reply:
I implemented it like this and remapped the IRQs on the STM32F303 DISCO BSP.